src/coord/src/coord.rs

```diff
@@ -2593,6 +2599,24 @@ impl Coordinator {
                     }
                     _ => (),
                 }
+
+                if let CatalogItem::Source(source) = item {
```

Comment:
This is more of a general comment, but I'm thinking where the logic for dropping belongs. If we look at purification, it has some similarities in that it also asynchronously goes out and queries external systems, it lives in the sql crate. However, the currently implemented purification steps don't have any side effects and that sounds like a useful property to keep.

I think what we want to describe here is async destroy logic for a `CatalogItem`, so what if just add this `async destroy()` method to it? In the `CatalogItem::Source` variant the implementation can forward the call to `SourceConnector::destroy` which then would call `ExternalSourceConnector::destroy` and eventually reach `PostgresSourceConnector::destroy`. This way we can keep the logic for a particular source close to its definition.

Reply:
Interesting -- I was personally deciding between this and something that looked like an equivalent to `purify`, but I like your suggestion! I'll swap this around.

Reply:
Hm. On second thought, even though I like this proposal, it seems a bit out of place to add side effects to the `dataflow-types` crate. 

@benesch, do you have thoughts either direction?